src/init/vyos-router

```diff
@@ -101,6 +101,16 @@ load_bootfile ()
     )
 }
 
+# restore if missing pre-config script
+restore_if_missing_preconfig_script ()
+{
+    if [ ! -x $vyatta_sysconfdir/config/scripts/vyos-preconfig-bootup.script ]; then
+        cp /usr/lib/live/mount/rootfs/*/opt/vyatta/etc/config/scripts/vyos-preconfig-bootup.script $vyatta_sysconfdir/config/scripts/
```

Comment:
Why copy each file across all installed images to the current one?

Reply:
Because according to the documentation these files (preconfig- and postconfig-script) are part of VyOS:

https://docs.vyos.io/en/latest/automation/command-scripting.html

By default each file is empty (except for the comment of the purpose of each file).

If user removes them then they should be recreated (empty) upon next reboot.

If the files already exists at destination then they wont be overwritten.

This fix also fixes the fact that preconfig-script never shows up after a system upgrade (add system image) since the preconfig-code was previously broken (it was called for in the init-script but the file never existed which this fixes that case aswell).

The use of wildcard is because the empty pre-/postconfig-script should exist in all installed images (this is the rootfs aka the squashfs file from the iso which is being referenced) so which one is selected doesnt matter primarly.

But sure if there is some way to select the last available then this would be prefered.

Also there is missing a environment variable that points to current rootfs so if that can be setup then of course that env variable can be used instead.

Reply:
> If user removes them then they should be recreated (empty) upon next reboot.

I disagree, they aren't essential - if they're missing, nothing executes. They can be added manually if needed.

> The use of wildcard is because the empty pre-/postconfig-script should exist in all installed images (this is the rootfs aka the squashfs file from the iso which is being referenced) so which one is selected doesnt matter primarly.

However an old script from a previous image could then be copied to the newly installed image and execute commands that are no longer relevant/could harm the current configuration.

If you wanted to guarantee the file is created empty, you could instead use `touch <path>` to generate the file.

Reply:
I aggree - but we should not only `touch(1)` the file if it‘s missing, but also create it with an appropriate note at the top.

Something like:

```bash
FOOFILE=$vyatta_sysconfdir/config/scripts/vyos-preconfig-bootup.script
if [ ! -x $FOOFILE ]; then
    cat << EOF > $FOOFILE
#!/bin/sh
# VyOS preconfig script
# add content here foo - bar baz
    EOF
    chmod +x $FOOFILE
fi
```

Potentially we could remove the entire template from the ISO script, too and just rely on this line to create the script in any case.

Reply:
@sarthurdev They are essential since they are mentioned in the docs regarding on how to perform post- and preconfig scripts. So if the native files are missing they should be recreated from the original "empty" ones.

Since the files are created by /src/debian/vyos-1x.postinst and are part of the squashfs they should be copied instead of having two places where the content of the files exists. This way the docs also match the reality.

@c-po I dont think the content of the files should be different, it should match the headers as displayed in the docs (https://docs.vyos.io/en/latest/automation/command-scripting.html). If something then a logging entry (also pushed to console) could be performed.

Reply:
I have created following feature request regarding current rootfs dir path that would resolve the use of wildcard as with this pull request: https://vyos.dev/T5457

Reply:
PR created for T5457: https://github.com/vyos/vyatta-op/pull/65

Reply:
I have added an update to this PR2135: https://github.com/vyos/vyos-1x/pull/2135/commits/d4d8517e0b78c8eae5d0b5518b09a0c4cb594e0e

Changes to original PR:

1) Using ${vyos_rootfs_dir} (now available thanks to merge of https://vyos.dev/T5457) instead of wildcard for if_missing_restore_preconfig and if_missing_restore_postconfig.

2) Adjusted chmod from 755 to 750 for the destination files (pre-/postconfig files copied to /config/scripts it missing).

Reply:
Hmm precious PR update needed a 2nd update but now I cant locate it :-S

Reply:
Found it!

This PR update2 contains following fixes compared to original PR2135:

https://github.com/vyos/vyos-1x/pull/2135/commits/2bfdfdfa80f9c351eaf57bffdc26ee772df8a59c

1) Using ${vyos_rootfs_dir} instead of wildcard for cp-operation in restore_if_missing_preconfig_script and restore_if-missing_postconfig_script.

2) Changed chmod from 755 to 750 for the files copied to destination.

3) Added proper brackets ${} for variables used by these two functions.